metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -151,6 +150,13 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 				},
 			}
 
+			hostSystem, err := getHostSystem(ctx, c, vm.Summary.Runtime.Host.Reference())
+			if err != nil {
+				logp.Debug("vsphere", err.Error())
+			} else {
+				event["hostname"] = hostSystem.Summary.Config.Name
```

Comment:
Should we put this under the global `host.name` field? See in ECS: https://github.com/elastic/ecs#host

It will require a bit more changes as with the `Fetcher` used here a few tricks must be applied to add fields on the root. I can help with that.

Reply:
I think I can do it..But  @bogd raised a valid concern.Maybe somebody is already using this field  as it is.I do not know what is the current state for it right now (is vsphere only in beta? )

Reply:
Yes, vsphere is still in beta to exactly allow such changes. I assume you are thinking of moving it for both metricsets which would make querying easier. Would be great to have that.

Reply:
Based on the current ECS https://www.elastic.co/guide/en/ecs/current/ecs-host.html, it should be `host.id` and `host.hostname`.

Reply:
Sorry, please ignore my comment above. I just saw https://github.com/elastic/beats/pull/7213#issuecomment-473921005